extensions/resteasy-common/deployment/src/main/java/io/quarkus/resteasy/common/deployment/ResteasyCommonProcessor.java

```diff
@@ -187,6 +191,29 @@ JaxrsProvidersToRegisterBuildItem setupProviders(BuildProducer<ReflectiveClassBu
         return new JaxrsProvidersToRegisterBuildItem(providersToRegister, contributedProviders, useBuiltinProviders);
     }
 
+    private void checkProperConfigAccessInProvider(AnnotationInstance instance) {
+        List<AnnotationInstance> configPropertyInstances = instance.target().asClass().annotations()
+                .get(ResteasyDotNames.CONFIG_PROPERTY);
+        if (configPropertyInstances == null) {
+            return;
+        }
+        for (AnnotationInstance configPropertyInstance : configPropertyInstances) {
+            if (configPropertyInstance.target().kind() != AnnotationTarget.Kind.FIELD) {
+                continue;
+            }
+            FieldInfo field = configPropertyInstance.target().asField();
+            Type fieldType = field.type();
+            if (ResteasyDotNames.CDI_PROVIDER.equals(fieldType.name())) {
```

Comment:
`javax.enterprise.inject.Instance` is also legal... and it extends `Provider` ;-).

Reply:
Actually I think we should just propose `Instance` since the name doesn't class with the JAX-RS Provider. WDYT?

Reply:
Fixed